src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -963,12 +963,20 @@ export class NotebookModel extends Disposable implements INotebookModel {
 
 	private async updateKernelInfoOnKernelChange(kernel: nb.IKernel, kernelAlias?: string) {
 		await this.updateKernelInfo(kernel);
+		let aliasLanguageInfo: nb.ILanguageInfo;
 		this.kernelAliases.forEach(kernel => {
```

Comment:
While you're here - this could be refactored a bit to be clearer and shortcut execution when it finds an item : 

kernelAlias = this.kernelAliases.find(kernel => this._defaultLanguageInfo?.name === kernel.toLowerCase()) ?? kernelAlias`